packages/ui/src/components/DataStory/clients/WorkspacesApi.ts

```diff
@@ -0,0 +1,14 @@
+export interface Tree {
+  path: string;
+  type: 'file' | 'folder';
+  content?: string;
+  children?: Tree[];
+}
+
+export interface WorkspacesApi {
+  get: ({ path }) => Promise<Tree>
+  create: ({ path, tree }: { path: string, tree: Tree }) => Promise<Tree>;
+  update: ({ path, tree }: { path: string, tree: Tree }) => Promise<Tree>
+  destroy: ({ path }: { path: string }) => Promise<void>
+  move: ({ path, newPath }: { path: string, newPath: string}) => Promise<Tree>
+}
```

Comment:
Perhaps we could draw some inspiration from this page in the AWS documentation: https://docs.aws.amazon.com/AmazonS3/latest/API/API_ListObjectsV2.html#API_ListObjectsV2_ResponseSyntax

Reply:
Interesting! It would certainly be more robust 👍